server/src/main/java/tabletop/services/TournamentService.java

```diff
@@ -0,0 +1,115 @@
+package tabletop.services;
+
+import org.springframework.stereotype.Service;
+import tabletop.domain.exceptions.BadRequestException;
+import tabletop.domain.exceptions.ErrorInfo;
+import tabletop.domain.match.tournament.Pair;
+import tabletop.domain.match.tournament.Tournament;
+import tabletop.domain.match.tournament.TournamentFinalResult;
+import tabletop.domain.match.tournament.TournamentType;
+import tabletop.domain.match.tournament.swiss.SwissPlayerResult;
+import tabletop.domain.match.tournament.swiss.SwissTournamentProcess;
+import tabletop.domain.user.User;
+import tabletop.repositories.TournamentFinalResultRepository;
+import tabletop.repositories.match.tournament.TournamentRepository;
+
+import java.util.*;
+
+@Service
+public class TournamentService {
+
+    private TournamentRepository tournamentRepository;
+    private TournamentFinalResultRepository tournamentFinalResultRepository;
+    private SwissTournamentService swissTournamentService;
+
+    public TournamentService(
+            TournamentRepository tournamentRepository,
+            TournamentFinalResultRepository tournamentFinalResultRepository,
+            SwissTournamentService swissTournamentService
+    ) {
+        this.tournamentRepository = tournamentRepository;
+        this.tournamentFinalResultRepository = tournamentFinalResultRepository;
+        this.swissTournamentService = swissTournamentService;
+    }
+
+    public Tournament getTournamentById(Long tournamentId) {
+        return tournamentRepository
+                .findOneById(tournamentId)
+                .orElseThrow(() -> new BadRequestException(ErrorInfo.TOURNAMENT_NOT_FOUND));
```

Comment:
I like this approach with orElseThrow but I think this shouldn't be done in the service layer but in the controller layer instead. Services can use each other and it wouldn't make sense to return a bad request exception when there was no request at all.